.github/workflows/test-backend.yml

```
@@ -38,19 +50,46 @@ jobs:
+          /usr/local/bin/ffmpeg
+          /usr/local/bin/ffprobe
+        # daily cache
+        key: ${{ runner.os }}-ffmpeg-${{ steps.current-date.outputs.today }}
+        restore-keys: |
+          ${{ runner.os }}-ffmpeg-${{ steps.current-date.outputs.today }}
     - name: Install FFmpeg
-      uses: FedericoCarboni/setup-ffmpeg@v3
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v4.0.3
+      if: steps.cache-ffmpeg.outputs.cache-hit != 'true' || github.event.inputs.force_ffmpeg_cache_update == true
+      run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC2015:info:7:41: Note that A && B || C is not if-then-else. C may run when A is true

(shellcheck)